x/auth/ante/unordered.go

```diff
@@ -76,3 +92,44 @@ func (d *UnorderedTxDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, _ bool, ne
 
 	return next(ctx, tx, false)
 }
+
+// txIdentifier returns a unique identifier for a transaction that is intended to be unordered.
+func txIdentifier(timeout uint64, tx sdk.Tx) ([32]byte, error) {
+	feetx := tx.(sdk.FeeTx)
+	if feetx.GetFee().IsZero() {
+		return [32]byte{}, errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "unordered transaction must have a fee")
+	}
+
+	buf := bufPool.Get().(*bytes.Buffer)
+	// Make sure to reset the buffer
+	buf.Reset()
+
+	// Use the buffer
+	for _, msg := range tx.GetMsgs() {
+		// loop through the messages and write them to the buffer
+		// encoding the msg to bytes makes it deterministic within the state machine.
+		// Malleability is not a concern here because the state machine will encode the transaction deterministically.
+		bz, err := proto.Marshal(msg)
+		if err != nil {
+			return [32]byte{}, errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "failed to marshal message")
+		}
+
+		buf.Write(bz)
+	}
+
+	// write the timeout height to the buffer
+	if err := binary.Write(buf, binary.LittleEndian, timeout); err != nil {
+		return [32]byte{}, errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "failed to write timeout_height to buffer")
+	}
+
+	// write gas to the buffer
+	if err := binary.Write(buf, binary.LittleEndian, feetx.GetGas()); err != nil {
+		return [32]byte{}, errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "failed to write unordered to buffer")
+	}
+
+	txHash := sha256.Sum256(buf.Bytes())
```

Comment:
personal preference: not sure if it is handled somewhere else but this is a cpu and memory intensive operation. It would be fair to charge an extra amount of gas for the operation and data .

Reply:
good call, ill look at adding something